napari/layers/_layer_actions.py

```diff
@@ -36,8 +36,9 @@ def _duplicate_layer(ll: LayerList):
     from copy import deepcopy
 
     for lay in list(ll.selection):
-        new = deepcopy(lay)
-        new.name += ' copy'
+        data, state, type_str = lay.as_layer_data_tuple()
+        state["name"] = trans._('{name} copy', name=lay.name)
+        new = Layer.create(deepcopy(data), state, type_str)
```

Comment:
how deepcopy work with Zarr and Dask data? or there should be no option to duplicate such layers?

Reply:
i think that ultimately we should add a feature to leverage layer.source a bit better.  One could imagine a `layer.reload` method that uses the layer source to bring back the data as it was originally brought in (using the same reader plugin and path)... We could discuss what to do when there is no source, but i think that would be a generally very useful thing and would definitely facilitate serialization of zarr/dask stuff

Reply:
Seems to work fine for dask:

```python
In [1]: import dask.array as da

In [2]: arr0 = da.random.random((5, 5))

In [3]: from copy import deepcopy

In [4]: arr1 = deepcopy(arr0)

In [5]: arr1
Out[5]: dask.array<copy, shape=(5, 5), dtype=float64, chunksize=(5, 5), chunktype=numpy.ndarray>

In [6]: import dask

In [7]: dask.compute(arr1[0,0])
Out[7]: (0.49711817392955315,)

In [8]: dask.compute(arr0[0,0])
Out[8]: (0.49711817392955315,)
```

Reply:
yeah, actually that makes a lot of sense, since a dask object is by its nature a very easily serializable object (more so that actual arrays)